### PR TITLE
Pull enumerations from run_jobs.py into separate vocab.py module (#96)

### LINKED ISSUE #96: Enums from run_jobs.py should be in a separate module
The `ValidJobName` and `ValidDataSourceName` enumerations should be pulled out of `run_jobs.py` (possibly to a `vocab.py` module) so they can be used by different jobs' entry modules, as well as `run_jobs.py`, without creating a circular dependency.

### PR DESCRIPTION
**Pull request by ssciolla**
This PR pulls out the enumerations from `run_jobs.py` into a separate `vocab.py` module so that both `run_jobs.py` and job-specific modules can reference the enumeration without introducing a circular dependency. It also changes the format of the `data_source` dictionaries returned from job entry functions so that the value associated with the key `data_source_name` is a member of `ValidDataSourceName`. This PR aims to resolve #96.

### CODE REVIEW

File: vocab.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# standard libraries
+from enum import auto, Enum
+
+
+# Enum(s)
+
+# Each job name should be defined in ValidJobName.
+# NAME_OF_JOB = 'path.to.method'
```
Comment (lsloan) [inline vocab.py:8]: What do you think of moving comments like these into docstrings?

File: course_inventory/inventory.py [hunk 2]
```diff
@@ -180,7 +181,7 @@ def pull_sis_section_data_from_udw(section_ids: Sequence[int], conn: connection)
 
 # Entry point for run_jobs.py
 
-def run_course_inventory() -> Sequence[Dict[str, Union[str, pd.Timestamp]]]:
+def run_course_inventory() -> Sequence[Dict[str, Union[ValidDataSourceName, pd.Timestamp]]]:
```
Comment (lsloan) [inline course_inventory/inventory.py:184]: I like that the return type is no longer a mysterious, undescribed string.

**Comment by lsloan** [comment]
I changed the description's "resolve issue #96" to "resolve #96" to trigger GH's issue linking feature.

File: vocab.py (hunk 1 above)
Comment (ssciolla) [reply vocab.py]: I moved them, let me know if you see any issues with them.

**Pull request merged**